Relationships/RelationshipResolver.cs

```diff
@@ -1,284 +1,422 @@
-    {
-        public bool with_all_suggests;
-        public bool with_recommends = true;
-        public bool with_suggests;
-        public bool without_toomanyprovides_kraken = false;
-        public bool without_enforce_consistency = false;
-        public object Clone()
-        {
-            return MemberwiseClone();
-        }
-    }
-
-    // Alas, it appears that structs cannot have defaults. Try
-    // DefaultOpts() to get friendly defaults.
-
-    // TODO: RR currently conducts a depth-first resolution of requirements. While we do the
-    // right thing in processing all depdenencies first, then recommends, and then suggests,
-    // we could find that a recommendation many layers deep prevents a recommendation in the
-    // original mod's recommends list.
-    //
-    // If we resolved in things breadth-first order, we're less likely to encounter surprises
-    // where a nth-deep recommend blocks a top-level recommend.
-
-    public class RelationshipResolver
-    {
-        // A list of all the mods we're going to install.
-        private static readonly ILog log = LogManager.GetLogger(typeof(RelationshipResolver));
-        private readonly Dictionary<string, CkanModule> modlist = new Dictionary<string, CkanModule>();
-        private Registry registry;
-        private KSPVersion kspversion;
-
-        
+	public class RelationshipResolverOptions : ICloneable
+	{
+		public bool with_recommends = true;
+
+		public bool with_suggests;
+		//Recursively add suggests.
+		public bool with_all_suggests;
+
+		public bool without_toomanyprovides_kraken = false;
+		public bool without_enforce_consistency = false;
+		public bool procede_with_inconsistencies = false;
+
+		public object Clone()
+		{
+			return MemberwiseClone();
+		}
+	}
+
+	// TODO: RR currently conducts a depth-first resolution of requirements. While we do the
+	// right thing in processing all depdenencies first, then recommends, and then suggests,
+	// we could find that a recommendation many layers deep prevents a recommendation in the
+	// original mod's recommends list.
+	//
+	// If we resolved in things breadth-first order, we're less likely to encounter surprises
+	// where a nth-deep recommend blocks a top-level recommend.
+
+	// TODO: Add mechanism so that clients can add mods with relationshup other than UserAdded. 
+	// Currently only made to support the with_{} options. 
+	public class RelationshipResolver
+	{
+		// A list of all the mods we're going to install.
+		private static readonly ILog log = LogManager.GetLogger(typeof (RelationshipResolver));
+		private readonly Dictionary<string, CkanModule> modlist = new Dictionary<string, CkanModule>();
+
+		private readonly List<KeyValuePair<Module, Module>> conflicts =
+			new List<KeyValuePair<Module, Module>>();
+
+		private readonly Dictionary<Module, Relationship> reasons = new Dictionary<Module, Relationship>();
+		private readonly Registry registry;
```

Comment:
I'm showing my C# ignorance here, but how does a `readonly` field work for something like this? I'm guessing it *can* be assigned in the constructor, but *cannot* be reassigned after that?